cmd/shellenv: use Bash.

MikeMcQuaid opened the pull request:
This speeds up execution by 3x on my machine and the script is simple enough to warrant this.

jamescostian commented:
Glad you're taking over this part, sorry I was out so long lol

I think you have to delete `Library/Homebrew/test/cmd/shellenv_spec.rb`

Also, [you had mentioned adding `HOMEBREW_PREFIX`](
https://github.com/Homebrew/brew/pull/4849#issuecomment-419716764), something like this I assume:

```bash
echo "export HOMEBREW_PREFIX='$HOMEBREW_PREFIX'"
```

Did you decide against that addition?

vitorgalvao commented on Library/Homebrew/cmd/shellenv.sh:
```diff
@@ -0,0 +1,10 @@
+
+homebrew-shellenv() {
+  echo "export PATH=\"$HOMEBREW_PREFIX/bin:#{HOMEBREW_PREFIX}/sbin:\$PATH\""
+  echo "export MANPATH=\"$HOMEBREW_PREFIX/share/man:\$MANPATH\""
```
Using single quotes inside double quotes won’t prevent variable expansion but will avoid the need for escaping the double quotes. For example, this line would become `echo "export MANPATH='$HOMEBREW_PREFIX/share/man:\$MANPATH'"` which I find marginally clearer.

vitorgalvao commented on Library/Homebrew/cmd/shellenv.sh:
```diff
@@ -0,0 +1,10 @@
+#:    Tip: have your dotfiles eval the output of this command
+
+homebrew-shellenv() {
+  echo "export PATH=\"$HOMEBREW_PREFIX/bin:#{HOMEBREW_PREFIX}/sbin:\$PATH\""
```
Shouldn’t `#{HOMEBREW_PREFIX}` be `$HOMEBREW_PREFIX`?

vitorgalvao commented:
> Also, [you had mentioned adding `HOMEBREW_PREFIX`](
https://github.com/Homebrew/brew/pull/4849#issuecomment-419716764), something like this I assume:
>
> ```bash
> echo "export HOMEBREW_PREFIX='$HOMEBREW_PREFIX'"
> ```

That would make `HOMEBREW_PREFIX` be empty (if yet unset). Do you mean:

```bash
echo "export HOMEBREW_PREFIX='$(brew --prefix)'"
```

jamescostian commented:
> That would make HOMEBREW_PREFIX be empty (if yet unset)

Not quite - notice how the other lines of code in this PR all use `$HOMEBREW_PREFIX`. If you don't believe me:

1. Clone Mike's fork and checkout his branch
2. Open `Library/Homebrew/cmd/shellenv.sh` in your editor
3. Right before the `}`, add `echo "export HOMEBREW_PREFIX='$HOMEBREW_PREFIX'"`
4. Run `bin/brew shellenv`

For example, my machine prints:

```bash
export PATH='/Users/james/code/brew/brew/bin:/Users/james/code/brew/brew/sbin':"$PATH"
export MANPATH='/Users/james/code/brew/brew/share/man':"$MANPATH"
export INFOPATH='/Users/james/code/brew/brew/share/man':"$INFOPATH"
export HOMEBREW_PREFIX='/Users/james/code/brew/brew'
```

MikeMcQuaid commented:
> Glad you're taking over this part, sorry I was out so long lol

No worries!



> I think you have to delete `Library/Homebrew/test/cmd/shellenv_spec.rb` to get the tests to pass

👍 yep.



> Also I think there's a typo in the bash script where it exports PATH but says `#{HOMEBREW_PREFIX}` instead of `$HOMEBREW_PREFIX`

Fixed 👍 



> Did you decide against that addition?

Added now along with `HOMEBREW_CELLAR` and `HOMEBREW_REPOSITORY`.
